### PR TITLE
Blind peer tests for setKeys

### PR DESCRIPTION
**Pull request by volstas**
Assumptions:
- It is fine to re-add even very "old" cores.
   - As in, setKeys cycles through everything that was set before. So if you call setKeys multiple times it will re-add the combination of entire history, not just the most recent set.
   - In turn,  I did not test or try to guarantee the "freshness" of the options. Because same core can have different "options" in different blind peers, when we reapply the options in setKeys(), we do not know which options object we took
- It is safe to get a core/autobase from a client that is destroyed/suspended and re-add it over new set of keys.
   - I did not deeply check the actual logic or write tests for these edge cases. The assumption is that the internals handle this correctly.
- Some of the options are not isolated, as in to test if "announce: true" gets correctly replicated, would also need to do more setup with trusted blind peers, etc. So I did not write tests for these more complicated options, focusing only on pick/target and priority.
   - Another way to test it would be checking deep internals of blind-peering to confirm the internal info object. Would not need to do setup but instead deeply depend on internal API. But at the moment did not write any explicit tests for those options.
- Autobase tests are single writer. I thought that should be fine for the purpose of rebalance.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -72,6 +72,131 @@ test('client can use a blind-peer to add a core', async (t) => {
...
+  ])
+
+  // Sanity check that it is what we expect before keys change
+  t.ok(core1Results.indexOf(2) === core1Results.lastIndexOf(2), '1 blindPeer swarmed for core1')
+  t.ok(core1Results.indexOf(0) !== core1Results.lastIndexOf(0), '2 blindPeers not swarm for core1')
+
+  t.is(await getBlindPeerCoreLength(blindPeer1, coreKey2), 1, 'blindPeer1 swarmed for core2')
+  t.is(await getBlindPeerCoreLength(blindPeer2, coreKey2), 1, 'blindPeer2 swarmed for core2')
+  t.is(await getBlindPeerCoreLength(blindPeer3, coreKey2), 1, 'blindPeer3 swarmed for core2')
+  // todo: remove signal when test is expected to pass
```
Comment (HDegroote) [inline test/test.js]: remove todo's?

File: test/test.js [hunk 2]
```diff
@@ -72,6 +72,131 @@ test('client can use a blind-peer to add a core', async (t) => {
...
+
+test('client can migrate multiple cores to multiple blind-peers and preserve settings', async (t) => {
+  const { bootstrap } = await getTestnet(t)
+  const [blindPeer1, blindPeer2, blindPeer3, blindPeer4, blindPeer5, blindPeer6] =
+    await getBlindPeers(6)
+
+  const { core, swarm, store } = await setupCoreHolder(t, bootstrap)
+  const core2 = store.get({ name: 'core2' })
+  await core2.append('Block 0')
+  swarm.join(core2.discoveryKey)
```
Comment (HDegroote) [inline test/test.js]: Why is this needed?

**Review by HDegroote — APPROVED** [review]
looks good
